contracts/Pool.sol

```diff
@@ -349,6 +547,73 @@ abstract contract Pool {
         amtDeltas[thisCycle].thisCycle += thisCycleBlocks * perBlockDelta;
         amtDeltas[thisCycle].nextCycle += nextCycleBlocks * perBlockDelta;
     }
+
+    /// @notice Stops proxy of `msg.sender` for the duration of the modified function.
+    /// This removes and then restores any effects of the proxy on all of its receivers' futures.
+    /// It allows the function to safely modify any properties of the proxy
+    /// without having to updating the state of its receivers.
+    modifier suspendProxy {
+        applyProxyDeltasOnReceivers(-1);
+        _;
+        applyProxyDeltasOnReceivers(1);
+    }
+
+    /// @notice Applies the effects of the proxy on all of its receivers' futures.
+    /// Effects are applied as if the change was made on the beginning of the current cycle.
+    /// @param multiplier The multiplier of the deltas applied on the receivers.
+    /// `-1` to remove the effects of the proxy, `1` to reapply after removal.
+    function applyProxyDeltasOnReceivers(int8 multiplier) internal {
+        uint64 blockNumber = uint64(block.number);
+        Proxy storage proxy = proxies[msg.sender];
+        uint32 receiversCount = 0;
+        // Create an in-memory copy of the receivers list to reduce storage access
+        ReceiverWeight[] memory receiversList = new ReceiverWeight[](PROXY_WEIGHTS_COUNT_MAX);
+        // Iterating over receivers, see `ReceiverWeights` for details
+        address receiverAddr = ReceiverWeightsImpl.ADDR_ROOT;
+        while (true) {
+            uint32 weight;
+            (receiverAddr, weight, ) = proxy.receiverWeights.nextWeightPruning(receiverAddr);
+            if (receiverAddr == ReceiverWeightsImpl.ADDR_ROOT) break;
+            require(receiversCount < PROXY_WEIGHTS_COUNT_MAX, "Too many proxy receivers");
+            receiversList[receiversCount++] = ReceiverWeight(receiverAddr, weight);
+        }
+        // The proxy doesn't exist
+        if (receiversCount == 0) return;
+
+        // Iterating over deltas, see `ProxyDeltas` for details
+        uint64 cycle = ProxyDeltasImpl.CYCLE_ROOT;
+        uint64 finishedCycle = blockNumber / cycleBlocks;
+        uint64 currCycle = finishedCycle + 1;
+        int128 totalDelta = 0;
+        while (true) {
+            int128 thisCycleDelta;
+            int128 nextCycleDelta;
```

Comment:
How do you feel about using `cycleDelta` and `followingCycleDelta` to avoid confusion with the `currCycle`? The “this” prefix seems to to indicate that we’re talking about the current cycle.

Reply:
I think that overall having `this` and `next` is a clean and concise convention. It highlights that there are two deltas relative to a containing data structure key and to each other, but independent from the current block. I took a look in the code and general removal of `this` convention would require some wrestling with renaming of suddenly vague and meaningless variable names.